### ISSUE TITLE
Compile errors due to c++14_types.hpp

### ISSUE THREAD

**Issue by sbastrakov** [ComputationalRadiationPhysics/liFFT#27]
Creating an issue just to document the behaviour discovered at hackathon with @sk1p . 

With current compilers this file causes compile errors. And the way it tries to extend namespace std is technically incorrect. It seems the solution can be to just remove this file, replace its includes with <type_traits> and raise C++ standard requirements to C++14 (C++11 is mostly enough, but there is stuff like `std::result_of_t` that are 14).

**Comment by Flamefire** [comment]
Correct. If wanted the code can be wrapped in a conditional `#if not C++14`
